### ISSUE TITLE
Unrecognized custom error message not returning custom error selector

### ISSUE THREAD

**Issue by kevincheng96** [NomicFoundation/hardhat#3237]
Hi Hardhat team, I'm not seeing the custom error selector returned in the `unrecognized custom error` error message that Hardhat produces. I [originally responded](https://github.com/NomicFoundation/hardhat/pull/2973#issuecomment-1253972559) on the PR that implemented this feature as part of [version 2.10.2](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.10.2). I was asked to open up a new issue with reproduction steps, so I have listed the steps here:

https://github.com/compound-finance/comet/pull/592

**Comment by fvictorio** [comment]
Oh, and in that case, I think this is a duplicate of https://github.com/NomicFoundation/hardhat/issues/3091. Can you confirm if that's the case?

**Comment by kevincheng96** [comment]
> Oh, and in that case, I think this is a duplicate of #3091. Can you confirm if that's the case?

I'm not actually sure if it's the same issue as #3091, since we have been able to test custom errors in the past when the error was recognized. e.g. `    await expect(contract.doSomething()).to.be.revertedWith("custom error 'Unauthorized()'");`

Though I agree with you that the problem seems specific to the chai matchers error message. Are those error messages generated from `solidity-errors.ts` instead of `error-inferror.ts`?

We did create our own patch for this on a fork of Hardhat if you're curious: https://github.com/jflatow/hardhat/commit/8cf3340fe146e77e411b5854c47ebc4d61fa7ec8. Basically, we added the custom error selector to the error message in [`solidity-errors.ts`](https://github.com/NomicFoundation/hardhat/blob/5a9927dd2ecbb6bbe60b35972c5969981cc080f3/packages/hardhat-core/src/internal/hardhat-network/stack-traces/solidity-errors.ts#L274) since it was not present there.
